plugins/org.eclipse.elk.core.service/src/org/eclipse/elk/core/service/LayoutMetaDataService.java

```diff
@@ -286,6 +286,11 @@ public void register(final LayoutAlgorithmData algorithmData) {
         @Override
         public void register(final LayoutOptionData optionData) {
             layoutOptionMap.put(optionData.getId(), optionData);
+            if (optionData.getLegacyIds() != null) {
+                for (String legacyId : optionData.getLegacyIds()) {
+                    layoutOptionMap.put(legacyId, optionData);
```

Comment:
I wouldn't know from the top of my head if requesting a `LayoutOptionData` instance through a given ID and getting an instance that has another ID might be a problem for some clients of the meta data service. I guess you have checked all calls to the `getOptionData(...)` methods?